sbpy/photometry/core.py

```diff
@@ -5,13 +5,255 @@
-           'DiskFunctionModel', 'LommelSeeliger', 'Lambert',
+__all__ = ['ref2mag', 'mag2ref', 'spline',
+           'DiskIntegratedModelClass', 'HG', 'HG12', 'HG1G2',
+           'DiskFunctionModel', 'LommelSeeliger', 'Lambert', 'LunarLambert',
            'PhaseFunctionModel', 'ROLOPhase',
            'ResolvedPhotometricModelClass', 'ROLO']
 
+import numpy as np
+from scipy.integrate import quad
+from astropy.modeling import FittableModel, Fittable1DModel, Fittable2DModel, Parameter
+from astropy import units
+
+
+def ref2mag(ref, radius, M_sun=None):
+    """Convert average bidirectional reflectance to reduced magnitude
+
+    Parameters
+    ----------
+    ref : num, astropy.units.Quantity
+        Average bidirectional reflectance
+    radius : num, astropy.units.Quantity
+        Radius of object
+    M_sun : number, optional
+        The magnitude of the Sun, default is -26.74
+
+    Returns
+    -------
+    The same type as input, reduced magnitude (at r=delta=1 au)
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> mag = ref2mag(0.1, 460)
+    >>> print('{0:.4}'.format(mag))
+    2.078
+    >>> mag = ref2mag(0.1, 460*u.km)
+    >>> print('{0:.4}'.format(mag))
+    2.078 mag
+    """
+
+    if M_sun is None:
+        M_sun = -26.74
+    Q = False
+    if isinstance(ref, units.Quantity):
+        ref = ref.value
+        Q = True
+    if isinstance(radius, units.Quantity):
+        radius = radius.to('km').value
+        Q = True
+    if isinstance(M_sun, units.Quantity):
+        M_sun = M_sun.to('mag').value
+        Q = True
+
+    mag = M_sun-2.5*np.log10(ref*np.pi*radius*radius*units.km.to('au')**2)
+    if Q:
+        return mag*units.mag
+    else:
+        return mag
+
+
+def mag2ref(mag, radius, M_sun=None):
+    """Convert reduced magnitude to average bidirectional reflectance
+
+    Parameters
+    ----------
+    mag : num, astropy.units.Quantity
+        Reduced magnitude
+    radius : num, astropy.units.Quantity
+        Radius of object
+    M_sun : number, optional
+        The magnitude of the Sun, default is -26.74
+
+    Returns
+    -------
+    The same type as input, average bidirectional reflectance
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> ref = mag2ref(2.08, 460)
+    >>> print('{0:.4}'.format(ref))
+    0.09981
+    >>> ref = mag2ref(2.08, 460*u.km)
+    >>> print('{0:.4}'.format(ref))
+    0.09981 1 / sr
+    """
+
+    if M_sun is None:
+        M_sun = -26.74
+    Q = False
+    if isinstance(mag, units.Quantity):
+        mag = mag.value
+        Q = True
+    if isinstance(radius, units.Quantity):
+        radius = radius.to('km').value
+        Q = True
+    if isinstance(M_sun, units.Quantity):
+        M_sun = M_sun.to('mag').value
+        Q = True
+
+    ref = 10**((M_sun-mag)*0.4)/(np.pi*radius*radius*units.km.to('au')**2)
+    if Q:
+        return ref/units.sr
+    else:
+        return ref
+
+
+class spline(object):
```

Comment:
Not really.  `scipy.interpolate.spline` doesn't work for this case, because in the G1G2 model the spline interpolation is defined with also the first derivatives at both ends, which is not in any spline implementation in Python that I could find.

Reply:
ok, sounds good!